src/main/java/org/broadinstitute/hellbender/utils/haplotype/HaplotypeBAMDestination.java

```diff
@@ -3,29 +3,46 @@
 import htsjdk.samtools.SAMFileHeader;
 import htsjdk.samtools.SAMProgramRecord;
 import htsjdk.samtools.SAMReadGroupRecord;
+import java.nio.file.Path;
+import org.broadinstitute.hellbender.utils.read.SAMFileGATKReadWriter;
+
 
 import org.broadinstitute.hellbender.utils.Utils;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
 
 import java.util.ArrayList;
 import java.util.List;
 
 /**
- * Utility class that allows easy creation of destinations for the HaplotypeBAMWriters
+ * Class used to direct output from a HaplotypeBAMWriter to a bam/sam file.
  *
  */
-public abstract class HaplotypeBAMDestination {
+
+public class HaplotypeBAMDestination {
+    private final SAMFileGATKReadWriter samWriter;
     private final SAMFileHeader bamOutputHeader;
     private final String haplotypeReadGroupID;
     private final static String haplotypeSampleTag = "HC";
 
     /**
      * Create a new HaplotypeBAMDestination
      *
-     * @param sourceHeader SAMFileHeader used to seed the output SAMFileHeader for this destination.
-     * @param haplotypeReadGroupID read group ID used when writing haplotypes as reads
+     * @param outPath path where output is written (doesn't have to be local)
+     * @param createBamOutIndex true to create an index file for the bamout
+     * @param createBamOutMD5 true to create an md5 file for the bamout
+     * @param sourceHeader SAMFileHeader used to seed the output SAMFileHeader for this destination, must not be null
+     * @param haplotypeReadGroupID  read group ID used when writing haplotypes as reads
      */
-    protected HaplotypeBAMDestination(SAMFileHeader sourceHeader, final String haplotypeReadGroupID) {
+    protected HaplotypeBAMDestination(
+            final Path outPath,
+            final boolean createBamOutIndex,
+            final boolean createBamOutMD5,
+            final SAMFileHeader sourceHeader,
+            final String haplotypeReadGroupID)
+    {
+        Utils.nonNull(outPath, "outputPath cannot be null");
+        Utils.nonNull(sourceHeader, "sourceHeader cannot be null");
```

Comment:
I think you're checking `sourceHeader` twice.

Reply:
Yes, you're right. I will remove the second instance of this.